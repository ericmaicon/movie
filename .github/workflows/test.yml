name: Lint & Test

on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: '14'
      - name: Cache dependencies
        uses: actions/cache@v2.1.4
        id: cache-dependencies
        with:
          path: |
            node_modules
            */*/node_modules
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
      - name: Install Dependencies
        if: steps.cache-dependencies.outputs.cache-hit != 'true'
        run: yarn install && yarn bootstrap
      - name: Link
        run: yarn lerna link
      - name: Build deps
        run: yarn build
      - name: Lint
        run: yarn lint
      - name: Run Tests
        run: yarn test
      - uses: codecov/codecov-action@v1
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: packages/**/coverage
          fail_ci_if_error: true
