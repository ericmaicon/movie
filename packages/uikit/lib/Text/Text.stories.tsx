import React from 'react';

import { Story, Meta } from '@storybook/react';

import Text, { TextProps } from './Text';

export default {
  title: 'Typography/Text',
  component: Text,
} as Meta;

const Template: Story<TextProps> = (args) => <Text {...args} />;

export const Text1 = Template.bind({});
Text1.args = {
  children: 'Text 1',
};
