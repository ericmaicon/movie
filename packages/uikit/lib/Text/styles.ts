import styled from 'styled-components';

export const Text = styled.p`
  font-size: 0.2rem;
  ${({ theme }) => theme.typography.text};

  //tablet
  @media (min-width: 768px) and (max-width: 1024px) {
    font-size: 0.5rem;
  }

  //desktop
  @media (min-width: 1025px) {
    font-size: 1rem;
  }
`;
