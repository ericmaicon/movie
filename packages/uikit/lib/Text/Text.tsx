import React, { BaseHTMLAttributes, ReactNode } from 'react';

import * as Styled from './styles';

export interface TextProps extends BaseHTMLAttributes<HTMLParagraphElement> {
  /**
   * Text content.
   */
  children: ReactNode;
}

const Text = ({ children, ...rest }: TextProps) => {
  return <Styled.Text {...rest}>{children}</Styled.Text>;
};

export default Text;
