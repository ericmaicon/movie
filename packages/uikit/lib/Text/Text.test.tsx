import React from 'react';

import { render } from '~/test/index';

import Text from './Text';

describe('Text', () => {
  it('should render correctly', () => {
    const { getByText } = render(<Text>Text</Text>);

    const textElement = getByText('Text');

    expect(textElement).toBeInTheDocument();
    expect(textElement).toHaveTextContent(/^Text$/);
    expect(textElement).toMatchSnapshot();
  });
});
