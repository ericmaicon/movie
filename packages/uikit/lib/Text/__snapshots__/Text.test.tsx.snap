// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
.c0 {
  font-size: 0.2rem;
  font-family: Roboto;
  font-weight: 300;
  font-size: 1rem;
  line-height: 1em;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
}

@media (min-width:768px) and (max-width:1024px) {
  .c0 {
    font-size: 0.5rem;
  }
}

@media (min-width:1025px) {
  .c0 {
    font-size: 1rem;
  }
}

<p
  class="c0"
>
  Text
</p>
`;
